Fix truffle-hdwallet-provider webpack warnings

Got this when upgrading packages in **truffle-hdwallet-provider**:

```
$ npx webpack --config webpack/webpack.config.js
Hash: 77376e831f90d59409ba
Version: webpack 4.24.0
Time: 47333ms
Built at: 2018-12-10 12:14:14
       Asset      Size  Chunks             Chunk Names
    index.js  2.05 MiB       0  [emitted]  main
index.js.map  6.37 MiB       0  [emitted]  main
Entrypoint main = index.js index.js.map
  [1] external "path" 42 bytes {0} [built]
  [8] external "crypto" 42 bytes {0} [built]
 [11] external "util" 42 bytes {0} [built]
 [14] external "url" 42 bytes {0} [built]
 [15] external "assert" 42 bytes {0} [built]
 [16] external "fs" 42 bytes {0} [built]
 [18] external "buffer" 42 bytes {0} [built]
 [19] (webpack)/buildin/module.js 497 bytes {0} [built]
 [21] external "events" 42 bytes {0} [built]
 [25] external "http" 42 bytes {0} [built]
 [56] external "os" 42 bytes {0} [built]
[176] external "child_process" 42 bytes {0} [built]
[319] ./src/index.js 5.03 KiB {0} [built]
[381] external "tty" 42 bytes {0} [built]
[382] external "net" 42 bytes {0} [built]
    + 883 hidden modules

WARNING in /Users/gnidan/src/work/truffle/node_modules/bindings/bindings.js 81:22-40
Critical dependency: the request of a dependency is an expression
 @ /Users/gnidan/src/work/truffle/node_modules/keccak/bindings.js
 @ /Users/gnidan/src/work/truffle/node_modules/keccak/index.js
 @ /Users/gnidan/src/work/truffle/node_modules/ethereumjs-util/dist/index.js
 @ ./src/index.js

WARNING in /Users/gnidan/src/work/truffle/node_modules/bindings/bindings.js 81:43-53
Critical dependency: the request of a dependency is an expression
 @ /Users/gnidan/src/work/truffle/node_modules/keccak/bindings.js
 @ /Users/gnidan/src/work/truffle/node_modules/keccak/index.js
 @ /Users/gnidan/src/work/truffle/node_modules/ethereumjs-util/dist/index.js
 @ ./src/index.js

WARNING in /Users/gnidan/src/work/truffle/node_modules/any-promise/register.js 24:14-37
Critical dependency: the request of a dependency is an expression
 @ /Users/gnidan/src/work/truffle/node_modules/any-promise/index.js
 @ /Users/gnidan/src/work/truffle/node_modules/web3-eth-accounts/src/index.js
 @ /Users/gnidan/src/work/truffle/node_modules/web3-eth/src/index.js
 @ /Users/gnidan/src/work/truffle/node_modules/web3/src/index.js
 @ ./src/index.js

WARNING in /Users/gnidan/src/work/truffle/node_modules/websocket/lib/BufferUtil.js
Module not found: Error: Can't resolve '../build/Release/bufferutil' in '/Users/gnidan/src/work/truffle/node_modules/websocket/lib'
 @ /Users/gnidan/src/work/truffle/node_modules/websocket/lib/BufferUtil.js
 @ /Users/gnidan/src/work/truffle/node_modules/websocket/lib/WebSocketFrame.js
 @ /Users/gnidan/src/work/truffle/node_modules/websocket/lib/websocket.js
 @ /Users/gnidan/src/work/truffle/node_modules/websocket/index.js
 @ /Users/gnidan/src/work/truffle/node_modules/web3/node_modules/web3-providers-ws/src/index.js
 @ /Users/gnidan/src/work/truffle/node_modules/web3/node_modules/web3-core-requestmanager/src/index.js
 @ /Users/gnidan/src/work/truffle/node_modules/web3/node_modules/web3-core/src/index.js
 @ /Users/gnidan/src/work/truffle/node_modules/web3/src/index.js
 @ ./src/index.js

WARNING in /Users/gnidan/src/work/truffle/node_modules/websocket/lib/Validation.js
Module not found: Error: Can't resolve '../build/Release/validation' in '/Users/gnidan/src/work/truffle/node_modules/websocket/lib'
 @ /Users/gnidan/src/work/truffle/node_modules/websocket/lib/Validation.js
 @ /Users/gnidan/src/work/truffle/node_modules/websocket/lib/WebSocketConnection.js
 @ /Users/gnidan/src/work/truffle/node_modules/websocket/lib/websocket.js
 @ /Users/gnidan/src/work/truffle/node_modules/websocket/index.js
 @ /Users/gnidan/src/work/truffle/node_modules/web3/node_modules/web3-providers-ws/src/index.js
 @ /Users/gnidan/src/work/truffle/node_modules/web3/node_modules/web3-core-requestmanager/src/index.js
 @ /Users/gnidan/src/work/truffle/node_modules/web3/node_modules/web3-core/src/index.js
 @ /Users/gnidan/src/work/truffle/node_modules/web3/src/index.js
 @ ./src/index.js

WARNING in /Users/gnidan/src/work/truffle/node_modules/websocket/lib/BufferUtil.js
Module not found: Error: Can't resolve '../build/default/bufferutil' in '/Users/gnidan/src/work/truffle/node_modules/websocket/lib'
 @ /Users/gnidan/src/work/truffle/node_modules/websocket/lib/BufferUtil.js
 @ /Users/gnidan/src/work/truffle/node_modules/websocket/lib/WebSocketFrame.js
 @ /Users/gnidan/src/work/truffle/node_modules/websocket/lib/websocket.js
 @ /Users/gnidan/src/work/truffle/node_modules/websocket/index.js
 @ /Users/gnidan/src/work/truffle/node_modules/web3/node_modules/web3-providers-ws/src/index.js
 @ /Users/gnidan/src/work/truffle/node_modules/web3/node_modules/web3-core-requestmanager/src/index.js
 @ /Users/gnidan/src/work/truffle/node_modules/web3/node_modules/web3-core/src/index.js
 @ /Users/gnidan/src/work/truffle/node_modules/web3/src/index.js
 @ ./src/index.js

WARNING in /Users/gnidan/src/work/truffle/node_modules/websocket/lib/Validation.js
Module not found: Error: Can't resolve '../build/default/validation' in '/Users/gnidan/src/work/truffle/node_modules/websocket/lib'
 @ /Users/gnidan/src/work/truffle/node_modules/websocket/lib/Validation.js
 @ /Users/gnidan/src/work/truffle/node_modules/websocket/lib/WebSocketConnection.js
 @ /Users/gnidan/src/work/truffle/node_modules/websocket/lib/websocket.js
 @ /Users/gnidan/src/work/truffle/node_modules/websocket/index.js
 @ /Users/gnidan/src/work/truffle/node_modules/web3/node_modules/web3-providers-ws/src/index.js
 @ /Users/gnidan/src/work/truffle/node_modules/web3/node_modules/web3-core-requestmanager/src/index.js
 @ /Users/gnidan/src/work/truffle/node_modules/web3/node_modules/web3-core/src/index.js
 @ /Users/gnidan/src/work/truffle/node_modules/web3/src/index.js
 @ ./src/index.js

WARNING in /Users/gnidan/src/work/truffle/node_modules/keccakjs/index.js
Module not found: Error: Can't resolve 'sha3' in '/Users/gnidan/src/work/truffle/node_modules/keccakjs'
 @ /Users/gnidan/src/work/truffle/node_modules/keccakjs/index.js
 @ /Users/gnidan/src/work/truffle/node_modules/ethereumjs-wallet/node_modules/ethereumjs-util/index.js
 @ /Users/gnidan/src/work/truffle/node_modules/ethereumjs-wallet/index.js
 @ ./src/index.js

WARNING in configuration
The 'mode' option has not been set, webpack will fallback to 'production' for this value. Set 'mode' option to 'development' or 'production' to enable defaults for each environment.
You can also set it to 'none' to disable any default behavior. Learn more: https://webpack.js.org/concepts/mode/
Done in 60.72s.
truffle-hdwallet-provider ✓
```

--- a/packages/truffle-hdwallet-provider/webpack/webpack.config.js
+++ b/packages/truffle-hdwallet-provider/webpack/webpack.config.js
@@ -5,6 +5,7 @@ const moduleRoot = resolve(__dirname, "..");
 const outputPath = join(moduleRoot, "dist");
 
 module.exports = {
+  mode: "production",
   entry: join(moduleRoot, "src", "index.js"),
   target: "node",
   devtool: "source-map",

--- a/packages/truffle-hdwallet-provider/webpack/webpack.config.js
+++ b/packages/truffle-hdwallet-provider/webpack/webpack.config.js
@@ -1,5 +1,6 @@
 const { resolve, join } = require("path");
 const { IgnorePlugin } = require("webpack");
+const path = require("path");
 
 const moduleRoot = resolve(__dirname, "..");
 const outputPath = join(moduleRoot, "dist");
@@ -16,7 +17,7 @@ module.exports = {
     libraryTarget: "umd",
     umdNamedDefine: true
   },
-  externals: ["fs"],
+  externals: ["fs", "bindings", "any-promise"],
   resolve: {
     alias: {
       // eth-block-tracker is es6 but automatically builds an es5 version for us on install. thanks eth-block-tracker!
@@ -26,7 +27,10 @@ module.exports = {
       "scrypt": "js-scrypt",
 
       // replace native `secp256k1` with pure js `elliptic.js`
-      "secp256k1": "secp256k1/elliptic.js"
+      "secp256k1": "secp256k1/elliptic.js",
+
+      // fix websocket require path
+      "websocket": path.resolve(__dirname, "../")
     }
   },
   plugins: [

